### ISSUE TITLE
Migrating to a new device should transfer vaccination certificates

### ISSUE THREAD

**Issue by odrotbohm** [corona-warn-app/cwa-app-ios#3625]
I just migrated to a new iOS device and used the local migration assistant that directly transfers apps, settings and content from device A to B via the local network, I assume. At least the other option is restoring from an iCloud backup.

I accidentally found out that the vaccination certificate was *not* transferred during that process, which might catch others by surprise, likely in a situation [when they actually have to present the certificate](https://twitter.com/manomama/status/1443695907988459521). That's arguably to be the worst situation to find out about that.

Is there a way to store the certificates in a way that they're part of an iOS backup or at least take part in the device to device migration?

**Comment by Ein-Tim** [comment]
Was other data of the CWA migrated to the new device @odrotbohm?

**Comment by Ein-Tim** [comment]
@odrotbohm 

Interesting that the local incidences were migrated. 
As far as I know, the current implementation is that there is no data migrated at all.

> That said, I don't consider that information critical. The vaccination certificate is, though.

I agree with you, but others might not, since it's easy to just scan a QR-Code in again, which is not possible for e.g. contact journal entries (they can't be imported, only exported). Same for check-ins, you would have to go back to the venue you checked into and scan the QR-Code.

And, shouldn't you know this: The random-IDs your old phone received from other phones are also not transferred, as they are stored by iOS and there in a part of the OS which is not backuped (or migrated, AFAIK).

**Comment by odrotbohm** [comment]
> Interesting that the local incidences were migrated.

I think you misread. They're gone, too.

> I agree with you, but others might not, since it's easy to just scan a QR-Code in again, which is not possible for e.g. contact journal entries (they can't be imported, only exported). Same for check-ins, you would have to go back to the venue you checked into and scan the QR-Code.

I totally see that the workaround for this is re-scanning the certificate, but having fully migrated (iOS migration assistant has sucessfully completed) the phone, I do not assume the certificate missing *unless I know about it not being part of the migration*. How should I know about this except by finding out accidentally, when I have to present the certificate (e.g. to gain entrance to a restaurant).

Also, folks tend to wipe their old phones after a successful migration. In that case I even have to have access to the original print of the QR code, which might not be the case.

In general, I backup my device. If I restore the device from a backup (local or from iCloud) I expect crucial data to be back on the device, because that's what I do backups for 🙃.

> And, shouldn't you know this: The random-IDs your old phone received from other phones are also not transferred, as they are stored by iOS and there in a part of the OS which is not backuped (or migrated, AFAIK).

I knew about this and also learned this in the device migration last year (read: also no active information about this in the app). It's less problematic though as those codes just serve me. The lack of them on the device doesn't prevent me from getting entrance to a concert or venue that I might have travelled quite some distance for, just to find out I can't access it as the certificate is gone.

Of course this is only an inconvenience, there is a workaround etc. So nothing terribly broken. Some entry in an FAQ might help already. Or an active notification screen when starting the app around the release of new iPhones ;).

**Comment by Ein-Tim** [comment]
> I think you misread. They're gone, too.

Yes, sorry, I misread that (:

> I totally see that the workaround for this is re-scanning the certificate, but having fully migrated (iOS migration assistant has sucessfully completed) the phone, I do not assume the certificate missing unless I know about it not being part of the migration. How should I know about this except by finding out accidentally, when I have to present the certificate (e.g. to gain entrance to a restaurant)?

I totally agree with you and see the problem here! But IHMO, the problem is "bigger" than the certificate "lost" issue. The app should clearly tell the user that no data is included in a backup and that one should export stored certificates as well as contact journal entries before switching to a new device. Best would be to tackle all these problems in one effort step (either making it possible to backup all data of the app (which can be backuped) or through creating a FAQ entry and linking to it from the app. 

> One final note why this probably catches folks by surprise: the app is not a "usual" app. You're not interacting with it on a day to day basis. I.e. the chance, that you actually have the chance to plug a message about that issue is low. Folks expect the app to just do "the right thing"™ in the background, and it blissfully does that usually. 🙃

Yeah, you're right. Maybe a good idea would be a splash screen (as you suggested) or add a new point to the "App information". Also, a push notification should be sent on the new device informing the user that the data of the app was not transferred (I will open an issue for this).

@dsarkar suggested in https://github.com/corona-warn-app/cwa-website/issues/1319 to extend the FAQ entry https://www.coronawarn.app/en/faq/#backup to cover more parts of the app. I strongly support this suggestion and also offered a PR.

**Comment by Jo-Achim** [comment]
In the event that the QR code of the digital vaccination card is not at hand, the FAQ entry should explicitly refer to the possibility of generating it as a PDF file within the CWA on the old smartphone and, if necessary, printing it out.

It seems possible to scan the digital vaccination certificate directly from the displayed PDF file (which means that no extra printout is necessary).

**Comment by odrotbohm** [comment]
If you have the old phone at hand you can just scan the QR code from the display, right? Maybe I am missing something but you start with "digital vaccination card is not at hand" but then conclude with "use the CWA", which you can only use to generate the PDF *if* the certificate is already in there, isn't it?

**Comment by Ein-Tim** [comment]
> If you have the old phone at hand you can just scan the QR code from the display, right?

Right.

**Comment by Jo-Achim** [comment]
This is, of course, even easier ;-).

**Comment by Ein-Tim** [comment]
@odrotbohm 

FYI: https://twitter.com/ThomasRgst/status/1460215789932457994?s=20 says that he used exactly the same method as you did & all data from CWA was there. 

I think the main issue to address is #3626, to align what is backed up and what not.

**Comment by michael-koeller** [comment]
> FYI: https://twitter.com/ThomasRgst/status/1460215789932457994?s=20 says that he used exactly the same method as you did & all data from CWA was there.

@Ein-Tim 

Not quite the exact method. I didn't use the migration assistant.
Instead, I did a local backup on my l MacBook connected via USB, then a restore from backup from the MacBook again via USB.
However, I doubt that this difference should have significant effect on what data is retained or not.

**Comment by Ein-Tim** [comment]
@odrotbohm 

Yeah, not exactly what you did, but what @odrotbohm, the user who initially reported this, did. 

Still, I agree with you, the data should be retained in any case.

**Comment by odrotbohm** [comment]
> @odrotbohm
> 
> FYI: https://twitter.com/ThomasRgst/status/1460215789932457994?s=20 says that he used exactly the same method as you did & all data from CWA was there.

There's multiple things that could be in play here:

- Has there been an update shipped that might have included the requested functionality?
- If you read closely, it's not clear whether the particular detail we discuss here is what the reply is talking about. It just states "Works by using the miration assistant." That's what I thought as well, until I needed the certificate. I.e. it's not clear whether he particularly responds to the "with content" part of the question.
- The problem still keeps being discussed in the tech podcast sphere, most recently in [The Freak Show](https://freakshow.fm/fs263-performance-performance-performance?t=1%3A09%3A53%2C1%3A23%3A22) and [Logbuch Netzpolitik](https://logbuch-netzpolitik.de/lnp412-kuss-auf-die-nuss?t=39%3A59%2C1%3A26%3A44) (both by Tim Pritlove).

**Comment by Ein-Tim** [comment]
@odrotbohm 

> Has there been an update shipped that might have included the requested functionality?

AFAIK not. 

> If you read closely, it's not clear whether the particular detail we discuss here is what the reply is talking about. It just states "Works by using the miration assistant." That's what I thought as well, until I needed the certificate. I.e. it's not clear whether he particularly responds to the "with content" part of the question.

I asked and the user confirmed that the same method as you used was used and that the certificate was there: https://twitter.com/thomasrgst/status/1460256404644765697?s=21

**Comment by Ein-Tim** [comment]
Another user, another result: https://twitter.com/Observer_de/status/1460272948149899266?s=20

**Comment by Ein-Tim** [comment]
I suggest to close this issue in favour of https://github.com/corona-warn-app/cwa-app-ios/issues/3626.

**Comment by dsarkar** [comment]
Hi @odrotbohm, please see https://github.com/corona-warn-app/cwa-app-ios/issues/3625#issuecomment-1098042816.

**Comment by stralau** [comment]
Re: the suggestion to scan certificates with the new phone from the old phone: this is usually too late as well, as the Apple migration workflow ends with erasing the old phone to prepare it for resale or recycling, and people won't notice before opening the CWA.

Also, I (and I suspect many others as well) don't have the paper vacc certificates anymore and need to get them reissued.

It would be nice to see this fixed, as it comes quite unexpected.
